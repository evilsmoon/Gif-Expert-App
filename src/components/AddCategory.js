import React, { useState } from 'react';
import PropTypes from "prop-types";

export const AddCategory = ({setCategories}) => {


    const [inputValue, setinputValue] = useState('');
    const handleInputChange = (e) => {
        setinputValue(e.target.value)
        // console.log('handleInputChange llamado');
    }
    const handelSubmit = (e) => {
        e.preventDefault();
        console.log('handelSubmit llamado ', inputValue);
        
        if(inputValue.trim().length >2){
            setCategories(c => [inputValue,...c]);
            setinputValue('');
        }
    }
    return (

        <form onSubmit={ handelSubmit }>  
            <p>{ inputValue }</p>
            <input
                type='text'
                value={ inputValue }
                onChange={ handleInputChange }
            />

        </form>
    )
}

AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired
}
