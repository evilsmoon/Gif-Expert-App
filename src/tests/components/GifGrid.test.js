import '@testing-library/jest-dom';
import React from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs');

describe('test GifGrid.js', () => {
    const category = 'One Piece';
    test('Debe de mostrat componente <GifGrid/> ', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    })
    test('debe demostrar items cuando se cargan imagenes usando useFetchGifs', () => {
        const gifs =[{
            id:'ABC',
            url:'https://localhost/cualquier/cosa.jpg',
            title:'cualquier cosa'
        }]
        
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })        
        const wrapper = shallow(<GifGrid category={category}/>);
        
        //    expect( wrapper ).toMatchSnapshot()
           expect( wrapper.find('p').exists()).toBe(false); 
           expect( wrapper.find('GifGridItem').length ).toBe( gifs.length ); 
    })
    
    
    
})
