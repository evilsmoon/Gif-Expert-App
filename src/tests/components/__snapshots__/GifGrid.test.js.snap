// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test GifGrid.js Debe de mostrat componente <GifGrid/>  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn "
  >
    One Piece
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
